### PR TITLE
Adding CITATION file

### PR DESCRIPTION
**Pull request by efajardo**
**Description of the Pull Request (PR):**

Adding a CITATION file per the new recommendations and per discussion on [Issue-1739](https://github.com/singularityware/singularity/issues/1739)

**This fixes or addresses the following GitHub issues:**

- Ref: # 1739

**Checkoff for all PRs:**

- [x] I have read the [Guidelines for Contributing](https://github.com/singularityware/singularity/blob/master/CONTRIBUTING.md), and this PR conforms to the stated requirements.
- [x] I have added changes to the [CHANGELOG](https://github.com/singularityware/singularity/blob/development/CHANGELOG.md) and and documentation updates to the [singularityware](https://www.github.com/singularityware/singularityware.github.io) documentation base.
- [ ] I have tested this PR locally with a `make test`
- [ ] This PR is NOT against the project's master branch
- [ ] I have added myself as a contributor to the [contributors's file](https://github.com/singularityware/singularity/blob/master/CONTRIBUTORS.md)
- [x] This PR is ready for review and/or merge

Attn: @singularityware-admin

### CODE REVIEW

**Comment by moranegg** [comment]
Hello,
Just a thought, the `repository-code` property should be the Github repository or another ongoing location of the code's development, because the archived version is already identified, here with the DOI.

This is helpful in linked data use cases, but also to find the maintained repository when there are many forks.

Another information, but might not be useful for a CFF file, is that the entire repository snapshot has been archived on Software Heritage with all branches and development history. You can use this link to access the last visited snapshot:
[swh:1:snp:9fef617df1de6298f504397c0cbc58ddee5db41a](https://archive.softwareheritage.org/swh:1:snp:9fef617df1de6298f504397c0cbc58ddee5db41a;origin=https://github.com/singularityware/singularity/)

For all visits made to this repository you have the visits view here: https://archive.softwareheritage.org/browse/origin/https://github.com/singularityware/singularity/visits/

Cheers,
Morane
Disclaimer: metadata specialist @ [Software Heritage](www.softwareheritage.org)

**Comment by efajardo** [comment]
Hi @moranegg ,

Well I am following the suggestions given by @dskatz. I used the automated tool [doi2cff](https://github.com/citation-file-format/doi2cff) for this to create the CFF.

File: CITATION.cff [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+# FIXME title as repository name might not be the best name, please make human readable
+title: Singularity 2.5.2 - Linux application and environment containers for science
+doi: 10.5281/zenodo.1308868
+# FIXME splitting of full names is error prone, please check if given/family name are correct
+authors:
+- given-names: Gregory
+  family-names: Kurtzer
+  name-particle: M et. al.
+  affiliation: Sylabs Inc.
+version: 2.5.2
```
Comment (GodloveD) [inline CITATION.cff:14]: Is there any way to automate this or just remove it?  I hate having a version number that we must increment every time we do a new release.

**Review by GodloveD — CHANGES_REQUESTED** [review]

File: CITATION.cff [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# YAML 1.2
+# Metadata for citation of this software according to the CFF format (https://citation-file-format.github.io/)
+cff-version: 1.0.3
+message: If you use this software, please cite it as below.
+# FIXME title as repository name might not be the best name, please make human readable
+title: Singularity 2.5.2 - Linux application and environment containers for science
+doi: 10.5281/zenodo.1308868
```
Comment (jmstover) [inline CITATION.cff:7]: This is a per version citation thing isn't it? So for 2.6.0 we would need a different `doi` ID right?

Comment (efajardo) [reply CITATION.cff:7]: @jmstover the doi will be created automatically by Zenodo but yes the CITATION file would have to be updated manually.

Comment (dtrudg) [reply CITATION.cff:7]: We can use the parent DOI in this case:

screen shot 2018-07-31 at 1 52 35 pm

**Comment by DrDaveD** [comment]
I think this PR is still viable and should be reopened.  The CITATION file can be updated to use the parent DOI, but the singularity release process would still need an extra step to include creating a new DOI for the new software version and updating the parent DOI to point to it.

**Comment by DrDaveD** [comment]
No I take it back; zenodo is supposed to be able to automatically create a new one with new github releases.  It hasn't happened yet for 2.6.0 however so something isn't quite right.

**Comment by dtrudg** [comment]
Annoyingly Zenodo *has* been creating versions.... but it didn't put them against the manually created doi for 2.5.2 - this wasn't clear from their docs.

They have all gone here.....  

10.5281/zenodo.1310023
https://doi.org/10.5281/zenodo.1310023

So the DOI needs to be changed again... :-( .... but now master is 3.0 this PR would need rebasing against that.

**Comment by GodloveD** [comment]
This was a great idea, but it doesn't seem like there is a lot of appetite for somebody to guide it to completion.  I'm going to close for now.  Might actually make sense to open a new PR if anybody wants to take another crack at it in the future.

**Pull request closed without merging**